torch/_dynamo/utils.py

```diff
@@ -146,6 +147,18 @@ def fmt_fn(values, item_fn=lambda x: x):
     torch.BoolTensor: (torch.bool,),
 }
 
+# TODO where should these live? circular import...
+# Control which distributed collectives and process group types are specially supported
+DYNAMO_SUPPORTED_COLLECTIVES = {
+    torch.distributed.all_reduce,
+}
+
+DYNAMO_SUPPORTED_PROCESS_GROUPS = {
+    torch.distributed.ProcessGroupGloo,
+}
+if hasattr(torch.distributed, "ProcessGroupNCCL"):
+    DYNAMO_SUPPORTED_PROCESS_GROUPS.add(torch.distributed.ProcessGroupNCCL)
+
```

Comment:
Just wanted to mention this feature is being worked on (https://github.com/pytorch/pytorch/issues/86225), with the main PR https://github.com/pytorch/pytorch/pull/88330 still in progress. This will change how ProcessGroups are initialized. `init_process_group()` will not longer set the default process group to be of type `torch.distributed.ProcessGroupNCCL` or `torch.distrbuted.ProcessGroupGloo` etc., but rather just `torch.distributed.ProcessGroup`. When a collective is called such as `dist.allreduce()` it will use `torch.distributed.ProcessGroup` internally route it to a C++ instance of `ProcessGroupGloo` / `ProcessGroupNCCL` based on the collective arguments. This change may cause an issue with PG matching.

Reply:
oh, thanks! i think that will only make things easier - i can just simplify this part to only support ProccessGroup - right?

Reply:
I think so, these changes arent specific to only gloo and nccl backend right? Will other lesser used ProcessGroup backends be supported such as ProcessGroupMPI, ProcessGroupUCC, or custom backends (e.g. XLA). If these changes are agnostic to that, then just `ProcessGroup` works!